zerver/lib/export.py

```diff
@@ -976,28 +984,71 @@ def get_ids(records: List[Record]) -> Set[int]:
         response['zerver_userprofile_mirrordummy'] +
         response['zerver_userprofile_crossrealm'])
 
-    if public_only:
+    consented_user_ids = set()  # type: Set[int]
+    if consent_message_id is not None:
+        consented_user_ids = get_consented_user_ids(consent_message_id)
+
+    if len(consented_user_ids) != 0:
+        public_streams = Stream.objects.filter(realm=realm, invite_only=False)
+        public_stream_recipient_ids = Recipient.objects.filter(
+            type=Recipient.STREAM, type_id__in=public_streams).values_list("id", flat=True)
+
+        consented_recipient_ids = Subscription.objects.filter(user_profile__id__in=consented_user_ids). \
+            values_list("recipient", flat=True)
```

Comment:
This is probably better written as `.values_list("recipient_id", flat=True)` -- otherwise it's sorta unclear whether we're expecting Recipient objects or IDs.